privacysandbox/topics_test.go

```diff
@@ -0,0 +1,594 @@
+				},
+				{
+					SegTax:   601,
+					SegClass: "2",
+					SegIDs:   []int{1},
+				},
+			},
+		},
+		{
+			name: "header with one valid field and another invalid field (w/o segIDs), should return only one valid field",
+			args: args{secBrowsingTopics: "(1);v=chrome.1:2:3, ();v=chrome.1:2:3, ();p=P0000000000"},
+			want: []Topic{
+				{
+					SegTax:   601,
+					SegClass: "3",
+					SegIDs:   []int{1},
+				},
+			},
+		},
+		{
+			name: "header with two valid fields having different model version",
+			args: args{secBrowsingTopics: "(1);v=chrome.1:2:3, (2);v=chrome.1:2:3, ();p=P0000000000"},
+			want: []Topic{
+				{
+					SegTax:   601,
+					SegClass: "3",
+					SegIDs:   []int{1},
+				},
+				{
+					SegTax:   601,
+					SegClass: "3",
+					SegIDs:   []int{2},
+				},
+			},
+		},
+		{
+			name: "header with one valid fields and two invalid fields (one with taxanomy < 0 and another with taxanomy > 10), should return only one valid field",
+			args: args{secBrowsingTopics: "(1);v=chrome.1:11:2, (1);v=chrome.1:5:6, (1);v=chrome.1:0:2, ();p=P0000000000"},
+			want: []Topic{
+				{
+					SegTax:   604,
+					SegClass: "6",
+					SegIDs:   []int{1},
+				},
+			},
+		},
+		{
+			name: "header with with valid fields having special characters (whitespaces, etc)",
+			args: args{secBrowsingTopics: "(1 2 4		6 7			4567	  ) ; v=chrome.1: 1 : 2, (1);v=chrome.1, ();p=P0000000000"},
+			want: []Topic{
+				{
+					SegTax:   600,
+					SegClass: "2",
+					SegIDs:   []int{1, 2, 4, 6, 7, 4567},
+				},
+			},
+		},
+		{
+			name: "header with one valid field having a few invalid segIds, should filter valid segIDs",
+			args: args{secBrowsingTopics: "(1 -3 9223372036854775808 0);v=chrome.1:1:2, ();p=P00000000000"},
+			want: []Topic{
+				{
+					SegTax:   600,
+					SegClass: "2",
+					SegIDs:   []int{1},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := ParseTopicsFromHeader(tt.args.secBrowsingTopics)
+			assert.Equal(t, tt.want, got, tt.name)
+		})
+	}
+}
+
+func TestUpdateUserDataWithTopics(t *testing.T) {
+	type args struct {
+		userData     []openrtb2.Data
+		headerData   []Topic
+		topicsDomain string
+	}
+	tests := []struct {
+		name string
+		args args
+		want []openrtb2.Data
+	}{
+		{
+			name: "empty topics, empty user data, no change in user data",
+			args: args{
+				userData:   nil,
+				headerData: nil,
+			},
+			want: nil,
+		},
+		{
+			name: "empty topics, non-empty user data, no change in user data",
+			args: args{
+				userData: []openrtb2.Data{
+					{
+						ID:   "1",
+						Name: "data1",
+						Segment: []openrtb2.Segment{
+							{ID: "1"},
+							{ID: "2"},
+						},
+					},
+				},
+				headerData: nil,
+			},
+			want: []openrtb2.Data{
+				{
+					ID:   "1",
+					Name: "data1",
+					Segment: []openrtb2.Segment{
+						{ID: "1"},
+						{ID: "2"},
+					},
+				},
+			},
+		},
+		{
+			name: "topicsdomain empty, no change in user data",
```

Comment:
Nitpick: could you update the name field so that `topicsdomain` -> `topicsDomain` matches the other variable in the test case.